embassy-stm32/src/adc/f1.rs

```diff
@@ -221,6 +234,75 @@ impl<'d, T: Instance> Adc<'d, T> {
         self.convert()
     }
 
+    pub async fn read_continuous<S, const N: usize, U>(
+        &mut self,
+        pin: &mut impl AdcPin<T>,
+        mut rxdma: PeripheralRef<'_, U>,
```

Comment:
I chosed to pass the DMA channel as a reference instead of sending it via the peripheral structure constructor.
As such, the dma channel can be re-used  safely for another peripheral once the transfer is done.
Sending a reference to the dma channel when constructing the peripheral structure means the channel is "bound forever" to the peripheral, which is not good in my opinion

Reply:
Another approach to that could be to drop the adc peripheral when you want to 'free' your channel for use by another peripheral. At least that's the approach taken for other peripherals afaict.

Reply:
Yeah but if you want to alternate dma transfer with two peripherals with the same channel, you constantly need to drop and re-construct them, which is I find not ideal.
But I think this situation would be kind of rare, so I'll change that to be homogenous with other peripherals.